build.rs

```diff
@@ -0,0 +1,6 @@
+/// Compile Protos in build-time
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
So as mentioned in our call – and I'll going to explain it in the code related to the version – here in `build.rs` you can use the `rustc_tools_util` crate to collect the information used in the binary:

```rust
    println!(
        "cargo:rustc-env=GIT_HASH={}",
        rustc_tools_util::get_commit_hash().unwrap_or_default()
    );
    println!(
        "cargo:rustc-env=COMMIT_DATE={}",
        rustc_tools_util::get_commit_date().unwrap_or_default()
    );
    println!(
        "cargo:rustc-env=RUSTC_RELEASE_CHANNEL={}",
        rustc_tools_util::get_channel().unwrap_or_default()
    );
```

Reply:
Added! Thanks for the snippet!